### PR TITLE
[RFC72] Update Localdb tests for Spring Boot Upgrade

### PR DESCRIPTION
**Pull request by inodb**
Related to https://github.com/cBioPortal/cbioportal/issues/10356

Only merge once the backend is approved&merged: https://github.com/cBioPortal/cbioportal/pull/10425

TODO:
- [x] nothing works yet :)
- [x] Somehow this shows up as a directory? `/tmp/repo/e2e-localdb-workspace/cbioportal-docker-compose/config/portal.properties`. It might be cleaner to run `cbioportal-docker-compose`'s `init.sh` and only replace any additional configuration by overriding through the `command:` in docker compose. That would avoid some duplication
- [x] WEB-INF folder has been removed in Spring Boot, how to handle?

### CODE REVIEW

File: end-to-end-test/local/docker_compose/keycloak.yml [hunk 1]
```diff
@@ -2,11 +2,11 @@ version: '3'
 
 services:
   cbioportal:
+    image: cbioportal/cbioportal:demo-rfc72
```
Comment (inodb) [inline end-to-end-test/local/docker_compose/keycloak.yml:5]: need to remove later, this is a hack b/c it's no longer picking up the `env/custom.sh` `BACKEND` parameter

File: env/custom.sh [hunk 2]
```diff
@@ -1,2 +1,3 @@
 #export CBIOPORTAL_URL="http://localhost:8081"
 #export GENOME_NEXUS_URL="https://www.genomenexus.org"
+BACKEND=cbioportal:demo-rfc72
```
Comment (alisman) [inline env/custom.sh:3]: @inodb did this work?  i tried this but coun't get it to work (in another PR).  I mean, couldn't even get it to checkout the right branch

Comment (inodb) [reply env/custom.sh:3]: this did not work, but updating `image:` directly does work

File: end-to-end-test/local/docker_compose/keycloak.yml [hunk 3]
```diff
@@ -2,11 +2,11 @@ version: '3'
...
     volumes:
-      - $TEST_HOME/docker_compose/keycloak/samlKeystore.jks:/cbioportal-webapp/WEB-INF/classes/samlKeystore.jks:ro
-      - $E2E_WORKSPACE/keycloak/idp-metadata.xml:/cbioportal-webapp/WEB-INF/classes/client-tailored-saml-idp-metadata.xml:ro
-    command: /bin/sh -c "java -Xms2g -Xmx4g -Dauthenticate=saml -Dsession.service.url=http://cbioportal-session:5000/api/sessions/my_portal/ -jar webapp-runner.jar -AmaxHttpHeaderSize=16384 -AconnectionTimeout=20000 --enable-compression /cbioportal-webapp"
-
+      - $TEST_HOME/docker_compose/keycloak/samlKeystore.jks:/cbioportal-webapp/samlKeystore.jks:ro
+      - $E2E_WORKSPACE/keycloak/idp-metadata.xml:/cbioportal-webapp/client-tailored-saml-idp-metadata.xml:ro
```
Comment (inodb) [inline end-to-end-test/local/docker_compose/keycloak.yml:8]: WEB-INF folder disappeared in spring boot

**Pull request merged**